content/blog/validation-split-as-3-way-split/index.Rmd

```diff
@@ -0,0 +1,151 @@
+
+childcare_costs <- childcare_costs |>
+  select(-matches("^mc_|^mfc")) |>
+  select(-county_fips_code) |>
+  na.omit() 
+
+glimpse(childcare_costs)
+```
+
+Even after omitting rows with missing values are we left with `r nrow(childcare_costs)` observations. That is plenty to work with! We are likely to get a reliable estimate of the model performance from a validation set without having to fit and evaluate the model multiple times, as with, for example, v-fold cross-validation.
+
+We are creating a three-way split of the data into a training, a validation, and a test set with the new `initial_validation_split()` function. We are stratifying based on our outcome `mcsa`. The default of `prop = c(0.6, 0.2)` means that 60% of the data gets allocated to the training set and 20% to the validation set - and the remaining 20% go into the test set.
+
+```{r initial-validation-split}
+set.seed(123)
+childcare_split <- childcare_costs |>
+  initial_validation_split(strata = mcsa)
+childcare_split
+```
+
+You can access the subsets of the data with the familiar `training()` and `testing()` as well as the new `validation()`:
+
+```{r validation}
+validation(childcare_split)
+```
+
+You may want to extract the training data to do some exploratory data analysis but here we are going to rely on xgboost to figure out patterns in the data so we can breeze straight to tuning a model.
+
+```{r xgb-wflow}
+xgb_spec <-
+  boost_tree(
+    trees = 500,
+    min_n = tune(),
+    mtry = tune(),
+    stop_iter = tune(),
+    learn_rate = 0.01
+  ) |>
+  set_engine("xgboost", validation = 0.2) |>
+  set_mode("regression")
+
+xgb_wf <- workflow(mcsa ~ ., xgb_spec)
+xgb_wf
+```
+
+We give this workflow object with the model specification to `tune_grid()` to try multiple combinations of the hyperparameters we tagged for tuning (`min_n`, `mtry`, and `stop_iter`). 
+
+<!-- But `tune_grid()` also needs data to fit the model on and data to assess the fitted model on! That's the training set and the validation set. The test set needs to be put aside until we've picked our final model. -->
+<!-- The new `validation_set()` function takes our initial three-way split object and returns the training set and the validation set, ready for use with the tuning functions like `tune_grid()`. -->
+
+During tuning, the model should not have access to the test data, only to the data used to fit the model (the analysis set) and the data used to assess the model (the assessment set). Each pair of analysis and assessment set forms a resample. With a validation split, we have just one resample with the training set functioning as the analysis set and the validation set as the assessment set. For 10-fold cross-validation, we'd have 10 resamples. The tidymodels tuning functions all expect a _set_ of resamples and the corresponding objects are of class `rset`. 
+To remove the test data from the initial three-way split and create such an `rset` object for tuning, use `validation_set()`.
```

Comment:
I've turned this over a few times and would love feedback on whether this is a clear motivation for the function name that also works for someone who is not too deeply immersed in `rsplit` vs `rset`.

Reply:
I think the sentence "For 10-fold cross-validation, we'd have 10 resamples." doesn't fit well. Maybe move it forward so this paragraph goes

- In tune we need analysis and assessment
- a pair of analysis and assessment forms a resample
- in 10-fold CV we have 10 resamples
- When using validation splits, training = analysis, validation = assessment
- thus validation splits gives us 1 resample

Reply:
Thanks! I think that works better 👍